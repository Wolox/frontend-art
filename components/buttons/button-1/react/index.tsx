import React from 'react';

import '../styles.scss';

function Button1() {
  return (
    <div className="column center btn-group-1">
      <button className="btn btn-1 m-bottom-2" disabled={true}>
        <div className="btn__content">cancel</div>
      </button>
      <button type="button" className="btn btn-1">
        <div className="btn__content">submit</div>
      </button>
    </div>
  );
}

export default Button1;
