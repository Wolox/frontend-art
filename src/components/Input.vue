<template lang="pug">
  .input-section-container
    #input-1.input-container
      span.input-subtitle Input 1
      .input-wrapper
        input.input-base(type='email')
        label.input-label Email
    #input-2.input-container
      span.input-subtitle Input 2
      label.inp(for='inp')
        input#inp(type='text' placeholder='Â ')
        span.label Label
        span.border
    #input-3.input-container
      span.input-subtitle Input 3
      .Input
        input#input.Input-text(type='text' placeholder='Your first name, e.g. Nicholas')
        label.Input-label(for='input') First name
    #input-4.input-container
      span.input-subtitle Input 4
      .Input
        .form-wrap
          .input-container
            label.label4(for='mail') Email address
            input.input4(type='email' name='mail' autocomplete='off')
          svg.line.svg4
</template>
<script>
import '../../scripts/input'
export default {

}
</script>

<style lang="scss" scoped>
@import './input/scss/input1';
@import './input/scss/input2';
@import './input/scss/input3';
@import './input/scss/input4';

.input-section-container {
  padding: 40px;
}

.input-container {
  display: flex;
  flex-direction: column;
  margin-bottom: 40px;
}

.input-subtitle {
  font-size: 14px;
  text-transform: uppercase;
  margin-bottom: 20px;
}
</style>

