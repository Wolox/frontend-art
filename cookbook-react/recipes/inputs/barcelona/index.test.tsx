import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, queryByAttribute }  from '@testing-library/react';

import Input from '.';

const getById = queryByAttribute.bind(null, 'data-testid');

describe('Barcelona Input', () => {
  const staticProps = { label: 'My Label', placeholder: 'Some Placeholder', name: "email"  }

  describe('when input is confirmable and there is no error', () => {
    const component = <Input {...staticProps} />;

    it('shows a valid input content', () => {
      const instance = TestRenderer.create(component)
      expect(instance.toJSON()!).toMatchSnapshot();
    })

    it('does not show the error message', () => {
      const { container } = render(component);
      expect(getById(container, 'email-error')?.classList).not.toContain('show');
    })

    it('shows the valid icon', () => {
      const { container } = render(component);
      expect(getById(container, 'email-valid-icon')).not.toBeNull;
    })

    it('shows the input as valid', () => {
      const { container } = render(component);
      expect(getById(container, 'email-input')?.classList).toContain('confirmed');
    })
  });

  describe('when input is confirmable and there is an error', () => {
    const component = <Input {...staticProps} error="Some error" />;

    it('shows an invalid input content', () => {
      const instance = TestRenderer.create(component)
      expect(instance.toJSON()!).toMatchSnapshot();
    })

    it('shows the error message', () => {
      const { container } = render(component);
      expect(getById(container, 'email-error')?.classList).toContain('show');
    })

    it('shows the valid icon', () => {
      const { container } = render(component);
      expect(getById(container, 'email-invalid-icon')).not.toBeNull;
    })
  });

  describe('when input is not confirmable and there is no error', () => {
    const component = <Input {...staticProps} confirmable={false} />;

    it('shows an invalid input content', () => {
      const instance = TestRenderer.create(component)
      expect(instance.toJSON()!).toMatchSnapshot();
    })

    it('does not show the error message', () => {
      const { container } = render(component);
      expect(getById(container, 'email-error')?.classList).not.toContain('show');
    })
  });

  describe('when input is not confirmable and there is an error', () => {
    const component = <Input {...staticProps} confirmable={false} error="Some error" />;

    it('shows an invalid input content', () => {
      const instance = TestRenderer.create(component)
      expect(instance.toJSON()).toMatchSnapshot();
    })
    
    it('shows the error message', () => {
      const { container } = render(component);
      expect(getById(container, 'email-error')?.classList).toContain('show');
    })

    it('shows the valid icon', () => {
      const { container } = render(component);
      expect(getById(container, 'email-invalid-icon')).not.toBeNull;
    })
  });

  describe('when the input is disabled', () => {
    const component = <Input {...staticProps} disabled={true} />;
    const { container } = render(component);

    it('shows the input as disabled', () => {
      expect(getById(container, 'email-input')?.getAttribute('disabled')).toBeTruthy;
    })
  })
})
