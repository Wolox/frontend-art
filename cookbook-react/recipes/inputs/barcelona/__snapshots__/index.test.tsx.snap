// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Barcelona Input when input is confirmable and there is an error shows an invalid input content 1`] = `
<div
  id="barcelona"
>
  <div
    className="column inputWrapper"
  >
    <label
      className="inputLabel m-bottom-1"
      htmlFor="barcelona-input"
    >
      My Label
    </label>
    <div
      className="inputContent"
    >
      <input
        className="inputBase m-bottom-1 inputError"
        data-testid="email-input"
        id="barcelona-input"
        name="email"
        placeholder="Some Placeholder"
      />
      <svg
        className="inputIcon"
        data-testid="email-valid-icon"
      >
        ic_invalid.svg
      </svg>
    </div>
    <span
      className="errorMessage show"
      data-testid="email-error"
    >
      Some error
    </span>
  </div>
</div>
`;

exports[`Barcelona Input when input is confirmable and there is no error shows a valid input content 1`] = `
<div
  id="barcelona"
>
  <div
    className="column inputWrapper"
  >
    <label
      className="inputLabel m-bottom-1"
      htmlFor="barcelona-input"
    >
      My Label
    </label>
    <div
      className="inputContent"
    >
      <input
        className="inputBase m-bottom-1 confirmed"
        data-testid="email-input"
        id="barcelona-input"
        name="email"
        placeholder="Some Placeholder"
      />
      <svg
        className="inputIcon"
        data-testid="email-valid-icon"
      >
        ic_valid.svg
      </svg>
    </div>
    <span
      className="errorMessage"
      data-testid="email-error"
    />
  </div>
</div>
`;

exports[`Barcelona Input when input is not confirmable and there is an error shows an invalid input content 1`] = `
<div
  id="barcelona"
>
  <div
    className="column inputWrapper"
  >
    <label
      className="inputLabel m-bottom-1"
      htmlFor="barcelona-input"
    >
      My Label
    </label>
    <div
      className="inputContent"
    >
      <input
        className="inputBase m-bottom-1 inputError"
        data-testid="email-input"
        id="barcelona-input"
        name="email"
        placeholder="Some Placeholder"
      />
      <svg
        className="inputIcon"
        data-testid="email-valid-icon"
      >
        ic_invalid.svg
      </svg>
    </div>
    <span
      className="errorMessage show"
      data-testid="email-error"
    >
      Some error
    </span>
  </div>
</div>
`;

exports[`Barcelona Input when input is not confirmable and there is no error shows an invalid input content 1`] = `
<div
  id="barcelona"
>
  <div
    className="column inputWrapper"
  >
    <label
      className="inputLabel m-bottom-1"
      htmlFor="barcelona-input"
    >
      My Label
    </label>
    <div
      className="inputContent"
    >
      <input
        className="inputBase m-bottom-1"
        data-testid="email-input"
        id="barcelona-input"
        name="email"
        placeholder="Some Placeholder"
      />
    </div>
    <span
      className="errorMessage"
      data-testid="email-error"
    />
  </div>
</div>
`;
